kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/storage/InMemoryKieServerTemplateStorage.java

```diff
@@ -54,17 +58,44 @@ public ServerTemplate store(ServerTemplate serverTemplate) {
 
     @Override
     public List<ServerTemplate> load() {
-        return new ArrayList<ServerTemplate>(store.values());
+        synchronized (store) {
```

Comment:
I guess we no longer need `store` to be of `ConcurrentHashMap` implementation if we are fully locking it.